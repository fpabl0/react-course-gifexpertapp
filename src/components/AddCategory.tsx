import { useState } from 'react';

type Props = {
  onNewCategory: (c: string) => void;
};

export const AddCategory = ({ onNewCategory }: Props) => {

  const [inputValue, setInputValue] = useState("");

  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(event.target.value);
  };

  const onFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const v = inputValue.trim();
    if (v.length <= 1) return;
    onNewCategory(v);
    setInputValue("");
  }

  return (
    <form onSubmit={onFormSubmit} aria-label="my-form">
      <input type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onChange} />
    </form>
  );
}
