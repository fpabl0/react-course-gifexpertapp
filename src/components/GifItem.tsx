import { Image } from '../helpers/getGifs';

type Props = Pick<Image, 'title' | 'url'>;

export const GifItem = ({ title, url }: Props) => {

  return (
    <div className='card'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
}
