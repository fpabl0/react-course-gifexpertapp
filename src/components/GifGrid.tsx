import { useEffect, useState } from 'react';
import { getGifs, Image } from "../helpers/getGifs";
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

type Props = {
  category: string;
};

export const GifGrid = ({ category }: Props) => {

  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {
        isLoading && (<h2>Cargando...</h2>)
      }

      <div className='card-grid'>
        {images.map(img => (
          <GifItem key={img.id}
            {...img} /> // this part spreads all the img fields as properties in the component
        ))}
      </div>
    </>
  );
}
