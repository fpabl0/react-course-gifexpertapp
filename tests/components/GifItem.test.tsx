import React from 'react'
import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Tests on <GifItem />', () => {

  const title = "Pikachu";
  const url = "https://www.pokemon.com/pikachu.gif"

  test('should match the snapshot', () => {

    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot();

  });

  test('should show the <img /> with the url and alt text given', () => {

    render(<GifItem title={title} url={url} />)
    // screen.debug();
    const img = screen.getByAltText<HTMLImageElement>(title);
    expect(img).toBeTruthy();
    expect(img.alt).toBe(title);
    expect(img.src).toBe(url);

  });

  test('should show the title in the component', () => {

    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title, { selector: 'p' })).toBeTruthy();

  });

});