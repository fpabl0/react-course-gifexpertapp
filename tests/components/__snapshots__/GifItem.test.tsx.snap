// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem /> should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pikachu"
      src="https://www.pokemon.com/pikachu.gif"
    />
    <p>
      Pikachu
    </p>
  </div>
</div>
`;
